src/NUnitFramework/framework/Attributes/ParallelizableAttribute.cs

```diff
@@ -51,6 +52,30 @@ public ParallelizableAttribute(ParallelScope scope) : base()
             Properties.Add(PropertyNames.ParallelScope, scope);
         }
 
+        /// <summary>
+        /// Overridden to check for invalid combinations of settings
+        /// </summary>
+        /// <param name="test"></param>
+        public override void ApplyToTest(Test test)
+        {
+            base.ApplyToTest(test);
+
+            if (test.RunState == RunState.NotRunnable)
+                return;
+
+            if (_scope.HasFlag(ParallelScope.Self) && _scope.HasFlag(ParallelScope.None))
+                MarkTestInvalid(test, "Test may not be both Parallel and NonParallel");
+
+            if (test is TestMethod && _scope.HasFlag(ParallelScope.ContextMask))
+                MarkTestInvalid(test, "ParallelScope of a test method may not specify Children or Fixtures");
+        }
+
+        private void MarkTestInvalid(Test test, string message)
```

Comment:
Rather than an extension method, I took the more direct approach of adding a MakeInvalid method to Test. It's only used in two places in this PR but could be used in over a dozen places elsewhere. I don't want to mess up this PR with that unrelated change, however, so it can be done later.